File: gestion-encuestas-backend/src/main/java/com/fullstack/encuestas/services/RespuestaService.java
package com.fullstack.encuestas.services;

import com.fullstack.encuestas.entities.Pregunta;
import com.fullstack.encuestas.entities.Respuesta;

import java.util.List;
import java.util.Optional;

public interface RespuestaService {

    List<Respuesta> getAllRespuestas();

    Optional<Respuesta> getRespuestaById(Long id);

    Respuesta createRespuesta(Respuesta respuesta);

    Respuesta updateRespuesta(Long id, Respuesta respuesta);

    void deleteRespuesta(Long id);

    List<Respuesta> findByPreguntaId(Long id);

    Respuesta agregarRespuestaAPregunta(Long id, Respuesta respuesta);
}

File: gestion-encuestas-frontend/src/app/services/respuesta.service.ts
import { Injectable } from '@angular/core';
import { Respuesta } from '../models/respuesta';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})

export class RespuestaService {

  private baseUrl:string='http://localhost:8080/api/respuestas';

  constructor(private httpClient:HttpClient) { }

  getAllRespuestas():Observable<Respuesta[]>{
    return this.httpClient.get<Respuesta[]>(this.baseUrl);
  }

  getRespuestaById(id:number):Observable<Respuesta>{
    return this.httpClient.get<Respuesta>(`${this.baseUrl}/${id}`);
  }

  updateRespuesta(id:number, Respuesta:Respuesta):Observable<Respuesta>{
    return this.httpClient.put<Respuesta>(`${this.baseUrl}/${id}`, Respuesta);
  }

  deleteRespuesta(id:number):Observable<void>{
    return this.httpClient.delete<void>(`${this.baseUrl}/${id}`);
  }

  saveRespuestaToPregunta(id: number, respuesta: Respuesta): Observable<Respuesta> {
    return this.httpClient.post<Respuesta>(`${this.baseUrl}/${id}`, respuesta);
  }

  getRespuestaByPreguntaId(id:number):Observable<Respuesta[]>{
    return this.httpClient.get<Respuesta[]>(`${this.baseUrl}/${id}/pregunta`);
  }
}
